src/libsystemd/sd-varlink/sd-varlink.c

```diff
@@ -3262,12 +3257,41 @@ static sd_varlink_server* varlink_server_destroy(sd_varlink_server *s) {
         sd_event_unref(s->event);
 
         free(s->description);
+        free(s->vendor);
+        free(s->product);
+        free(s->version);
+        free(s->url);
 
         return mfree(s);
 }
 
 DEFINE_PUBLIC_TRIVIAL_REF_UNREF_FUNC(sd_varlink_server, sd_varlink_server, varlink_server_destroy);
 
+_public_ int sd_varlink_server_set_info(
+                sd_varlink_server *s,
+                const char *vendor,
+                const char *product,
+                const char *version,
+                const char *url) {
+
+        assert_return(s, -EINVAL);
+
+        _cleanup_free_ char
+                *a = vendor ? strdup(vendor) : NULL,
+                *b = product ? strdup(product) : NULL,
+                *c = version ? strdup(version) : NULL,
+                *d = url ? strdup(url) : NULL;
+        if ((vendor && !a) || (product && !b) || (version && !c) || (url && !d))
```

Comment:
Maybe better to use `strdup_to()`.

Reply:
It comes out to about the same level of complication. We'd need to either collect the return values or use multiple checks. I think the current version is good enough…

Reply:
I do not think so, but at the same time, not a strong opinion. So, feel free to ignore.